compiler/semexprs.nim

```diff
@@ -873,6 +873,9 @@ proc evalAtCompileTime(c: PContext, n: PNode): PNode =
           localError(c.config, n.info, errCannotInterpretNodeX % renderTree(call))
         else: result = fixupTypeAfterEval(c, result, n)
       else:
+        let noneCtProc = sfCompileTime notin n[0].typ.owner.flags
+        if callee.kind == skConst and (n.typ == nil and noneCtProc):
+          return n
         result = evalConstExpr(c.module, c.idgen, c.graph, call)
```

Comment:
Why can't `evalConstExpr` evaluate procs?

Reply:
it fix `VM problem: dest register is not set` , here it try to eval at compile time, however the const closure is meant to run at runtime. there's no flag distinguish this, so it relys on `n.typ == nil`, yeah, after fix `fixupTypeAfterEval`, I may recheck here?

Reply:
just tried, still can't work without this. maybe should introduce new flag, make this condition simpler.